app/src/main/java/com/example/byitscover/scrapers/BarnesAndNobleScraper.java

```diff
@@ -43,50 +42,25 @@ public BarnesAndNobleScraper() {
      * @throws IOException
      */
     public List<BookListing> scrape(Query query) throws IOException {
-        List<Result> results = ScraperHelper.googleAPISearch(ScraperConstants.BARNES_AND_NOBLE, query);
-        String searchingUrl = getActualBookResult(results);
-        Document document = Jsoup.connect(searchingUrl).get();
-        System.out.println(searchingUrl);
+        String url = getUrlWithQuery(query);
+        System.out.println(url);
+        Document document = Jsoup.connect(url).get();
+        Elements bookLinks = document.getElementsByClass("pImageLink ");
 
-        Map<String, String> toReturn = new HashMap<String, String>();
+        List<BookListing> listings = new ArrayList<BookListing>();
 
-        //get rating by parsing json js var
-        String htmlString = document.html();
-        Double rating = null;
-        String reviewValueString;
+        BookListing firstListing = getListingFromElement(bookLinks, 0, query);
```

Comment:
Will the list of links always be nonempty? Maybe replace with

```
for (int i = 0; i < Math.min(3, bookLinks.size()) {
    listings.add(getListingFromElement(bookLinks, i, query));
}
```

Reply:
I like it, that's a clean way of doing it